apps/src/applab/AppLabTooltipOverlay.jsx

```diff
@@ -47,8 +47,8 @@ export class AppLabTooltipOverlay extends React.Component {
    * @private
    */
   getHoveredControlId(controlElement) {
-    // Check that the element is a child of a screen
-    if (!controlElement || $(controlElement).parents('div.screen').length === 0) {
+    // Check that the element is a child of designModeViz
+    if (!controlElement || $(controlElement).parents('div#designModeViz').length === 0) {
```

Comment:
This works correctly in design mode, but actually breaks this feature for screens _and_ all other elements in code mode.  I believe it should work in both (it's very helpful to hover and see element ids when coding).

Reply:
Glad I asked you! I had no idea it worked that way. Any suggestions on how to test that? If you can point me to anywhere we test tooltip overlays, that would help. (I'm currently spending about 20% of my time writing code and 80% of the my time trying to figure out how to get tests to work)

Reply:
The Circle CI failure on this PR is actually a unit test catching this exact issue.  The test is in [AppLabTooltipOverlayTest.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/applab/AppLabTooltipOverlayTest.js#L95-L117) - we've been trying to name test files consistently with source files for a while, so that's usually the first place to look.

The test in question is quite artificial, but it does check the desired behavior - it sets up a fake element on a fake `.screen`, then manually calls the `onMouseMove` handler for the overlay and checks the rendered set of tooltips.

```jsx
  describe('when over an applab control', () => {
    const CONTROL_ID = 'fake-id';

    beforeEach(() => {
      // Create an element on the screen
      let fakeScreen = document.createElement('div');
      fakeScreen.className = 'screen';

      let fakeElement = document.createElement('div');
      fakeElement.id = CONTROL_ID;
      fakeScreen.appendChild(fakeElement);

      let instance = ReactTestUtils.renderIntoDocument(<AppLabTooltipOverlay {...TEST_PROPS} />);
      instance.onMouseMove({ target: fakeElement });
      result = ReactTestUtils.findRenderedComponentWithType(instance, TooltipOverlay);
    });

    it('has a second provider that returns the element ID', () => {
      expect(result.props.providers.length).to.equal(2);
      expect(result.props.providers[1]).to.be.a('function');
      expect(result.props.providers[1](result.props)).to.equal(`id: ${CONTROL_ID}`);
    });
  });
```

Given the new behavior, I'd add a fake `#divApplab` (Code Mode) or `#designModeViz` (Design Mode) above the screen, and expand this into four tests:

|  | Code Mode | Design Mode |
| --- | --- | --- |
| `target: fakeElement` | Shows element ID | Shows element ID |
| `target: fakeScreen` | Shows screen ID | Shows screen ID |

Reply:
Thanks Brad - that was super helpful. I ended up with just two tests because my code modification no longer checks for the parent div ID (which was different between design and code mode), but instead checks for the parent div class `withCrosshair` (which is consistent between design and code mode).